src/lib/components/accordion/accordion.component.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useRef, useState } from 'react';
+
+import { Box } from '../box/Box';
+import { Icon } from '../icon/Icon.component';
+import { spacing, Stack } from '../../spacing';
+
+import styled from 'styled-components';
+
+import { Text } from '../text/Text.component';
+
+export type AccordionProps = {
+  title: string;
+  id: string;
+  children: React.ReactNode;
+  style?: React.CSSProperties;
+};
+
+const AccordionHeader = styled.button`
+  -webkit-appearance: none;
+  -moz-appearance: none;
+  appearance: none;
+  border: none;
+  gap: ${spacing.r8};
+  width: 100%;
+  cursor: pointer;
+  background-color: transparent;
+  color: ${(props) => props.theme.textPrimary};
+  padding: ${spacing.r4};
+  width: 100%;
+`;
+const AccordionContainer = styled.div<{
+  isOpen: boolean;
+  computedHeight: string;
+}>`
+  overflow: hidden;
+  opacity: ${(props) => (props.isOpen ? 1 : 0)};
+  height: ${(props) => props.computedHeight};
+  transition: height 0.3s ease-in, opacity 0.3s ease-in, visibility 0.3s;
+  visibility: ${(props) => (props.isOpen ? 'visible' : 'hidden')};
+`;
+const Wrapper = styled.div`
+  padding-block: ${spacing.r8};
+`;
+
+export const Accordion = ({ title, id, style, children }: AccordionProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const handleToggleContent = () => {
+    setIsOpen((prev) => !prev);
+  };
+  const calcContentHeight = () => {
+    if (containerRef.current && isOpen) {
+      const height = containerRef.current.scrollHeight;
+      return height + 'px';
+    } else return '0px';
+  };
+
+  return (
+    <Box style={{ width: '100%' }}>
+      <h3 style={{ margin: 0 }}>
+        <AccordionHeader
+          id={`Accordion header ${id}`}
+          onClick={handleToggleContent}
+          aria-controls={id}
+          aria-expanded={isOpen}
+          onKeyDown={(e) =>
+            (e.key === 'Enter' || e.key === ' ') && handleToggleContent
+          }
+        >
+          <Stack direction="horizontal" gap="r8">
+            <Icon
+              name="Chevron-up"
+              size="lg"
+              style={{
+                transform: isOpen ? 'rotate(0deg)' : 'rotate(180deg)',
+                transition: 'transform 0.3s ease-in',
+              }}
+            />
+            <Text isEmphazed>{title}</Text>
+          </Stack>
+        </AccordionHeader>
+      </h3>
+
+      <AccordionContainer
+        ref={containerRef}
+        computedHeight={calcContentHeight()}
```

Comment:
I wonder if we can simplify the code similarly to this way:
`ref={(element) => element.style.height = element.scrollHeight + 'px'}`